src/ResourceStream.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ZipStream;
+
+use Psr\Http\Message\StreamInterface;
+use RuntimeException;
+
+/**
+ * @internal
+ */
+class ResourceStream implements StreamInterface
+{
+    public function __construct(
+        /**
+         * @var resource
+         */
+        private $stream
+    ) {
+    }
+
+    public function __toString(): string
+    {
+        if ($this->isSeekable()) {
+            $this->seek(0);
+        }
+        return (string) stream_get_contents($this->stream);
+    }
```

Comment:
It's hard to tell, are we invoking these `__toString` methods in code?

We shouldn't be, as  a big stream, say 15gb could easily eat all of a users RAM. I suspect we aren't, but this method is dangerous none the less. I'd vote if we can eliminate the `__toString` from StreamInterface and it's implementers that'd be a net win.

Reply:
I had missed that StreamInterface was literally PSR-7 StreamInterface and not something from this library - and __toString is on that. Ignore my previous comment 😆